package.json

```diff
@@ -33,29 +34,41 @@
     "@types/time-span": "2.0.0",
     "async-retry": "1.2.3",
     "copyfiles": "2.3.0",
+    "conventional-changelog": "3.1.18",
+    "conventional-changelog-cli": "2.0.31",
+    "cz-conventional-changelog": "3.1.0",
     "husky": "1.3.1",
     "jasmine": "3.4.0",
     "jasmine-reporters": "2.3.2",
     "jasmine-spec-reporter": "4.2.1",
     "nyc": "15.0.1",
+    "pretty-quick": "2.0.1",
     "ts-node": "8.3.0",
     "tslint": "5.11.0",
     "tslint-config-standard": "8.0.1",
     "typescript": "3.9.6"
   },
   "scripts": {
+    "commit": "git-cz",
     "build": "tsc && copyfiles \"lib/**/*.json\" dist && copyfiles \"tests/**/*.js*\" dist && copyfiles \"package.json\" dist",
     "test": "npm run build && nyc jasmine --config=./tests/jasmine.json",
     "publish:unstable": "./scripts/publish-unstable.sh",
     "publish:release": "./scripts/publish.sh",
-    "version:release": "npm version --message \"release commit [skip ci]\"",
+    "version:release": "npm version --message \"chore(ref-imp): publish release [skip ci]\"",
     "lint": "tslint --fix --project .",
     "spec": "node ./scripts/spec-render.js",
-    "spec:edit": "node ./scripts/spec-edit.js"
+    "spec:edit": "node ./scripts/spec-edit.js",
+    "version": "conventional-changelog -p angular -i CHANGELOG.md -s && git add CHANGELOG.md"
   },
   "husky": {
     "hooks": {
-      "pre-commit": "npm run lint"
+      "pre-commit": "[[ -n $HUSKY_BYPASS ]] || npm run lint || pretty-quick --staged",
```

Comment:
what is pretty-quick --staged doing?

Reply:
I think automats the code in staged files.

Reply:
Good catch, nothing, will remove

Reply:
Oh? You didn't intend to use `pretty-quick` for auto formatting?

Reply:
We can, but have to install and configure prettier too e.g see here https://github.com/mattrglobal/bls12381-key-pair/blob/master/package.json#L35